op-e2e/e2eutils/fakebeacon/blobs.go

```diff
@@ -81,19 +81,47 @@ func (f *FakeBeacon) Start(addr string) error {
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
+
+		query := r.URL.Query()
+		rawIndices := query["indices"]
+		indices := make([]int, 0, len(bundle.Blobs))
+		if len(rawIndices) == 0 {
+			// request is for all blobs
+			for i := range bundle.Blobs {
+				indices = append(indices, i)
+			}
+		} else {
+			for _, raw := range rawIndices {
+				ix, err := strconv.ParseUint(raw, 10, 64)
```

Comment:
Since `ParseUint` will error if the raw index is sign-prefixed, the latter check that the index is negative should never be hit on line 110. Could we remove that check and just make the `indices` array of type `uint64`?

Reply:
That's certainly a reasonable approach. If a value's purpose is known to be an array index I prefer to convert to int at that point rather than deferring the conversion, but then as you note you have the awkward possibility of negative values that I always feel obliged to defensively check for (as done on L110).   I don't feel strongly about either approach TBH.